src/shred/shred.rs

```diff
@@ -0,0 +1,450 @@
+    opts.optflag("", "version", "output version information and exit");
+
+    let matches = match opts.parse(&args[1..]) {
+        Ok(m)  => m,
+        Err(e) => panic!("Invalid options\n{}", e)
+    };
+
+    if matches.opt_present("help") {
+        println!("Usage: {} [OPTION]... FILE...", NAME);
+        println!("Overwrite the specified FILE(s) repeatedly, in order to make it harder \
+                  for even very expensive hardware probing to recover the data.");
+        println!("{}", opts.usage(""));
+        println!("Delete FILE(s) if --remove (-u) is specified.  The default is not to remove");
+        println!("the files because it is common to operate on device files like /dev/hda,");
+        println!("and those files usually should not be removed.");
+        println!("");
+        return 0;
+    } else if matches.opt_present("version") {
+        println!("{} {}", NAME, VERSION_STR);
+        return 0;
+    } else if matches.free.is_empty() {
+        println!("{}: Missing an argument", NAME);
+        println!("For help, try '{} --help'", NAME);
+        return 0;
+    } else {
+        let iterations = match matches.opt_str("iterations") {
+            Some(s) => match s.parse::<usize>() {
+                           Ok(u) => u,
+                           Err(_) => {
+                               println!("{}: Invalid number of passes", NAME);
+                               return 1;
+                           }
+                       },
+            None => 3
+        };
+        let remove = matches.opt_present("remove");
+        let size = get_size(matches.opt_str("size"));
+        let exact = matches.opt_present("exact") && size.is_none(); // if -s is given, ignore -x
+        let zero = matches.opt_present("zero");
+        let verbose = matches.opt_present("verbose");
+        for path_str in matches.free.into_iter() {
+            wipe_file(&path_str, iterations, remove,
+                      size, exact, zero, verbose);
+        }
+    }
+    
+    0
+}
+
+// TODO: Add support for all postfixes here up to and including EiB
+//       http://www.gnu.org/software/coreutils/manual/coreutils.html#Block-size
+fn get_size(size_str_opt: Option<String>) -> Option<u64> {
+    if size_str_opt.is_none() { 
+        return None;
+    }
+    
+    let mut size_str = size_str_opt.as_ref().unwrap().clone();
+    // Immutably look at last character of size string
+    let unit = match size_str.chars().last().unwrap() {
+        'K' => { size_str.pop();  1024u64 }
+        'M' => { size_str.pop(); (1024 * 1024) as u64 }
+        'G' => { size_str.pop(); (1024 * 1024 * 1024) as u64 }
+         _   => { 1u64 }
+    };
+    
+    let coeff = match size_str.parse::<u64>() {
+        Ok(u) => u,
+        Err(_) => {
+            println!("{}: {}: Invalid file size", NAME, size_str_opt.unwrap());
+            exit!(1);
+        }
+    };
+    
+    Some(coeff * unit)
+}
+
+fn bytes_to_string(bytes: &[u8]) -> String {
```

Comment:
Looks like it's just parsing the size argument into a u64.  What part looks strange?